apps/condo/domains/ticket/schema/Ticket.js

```diff
@@ -333,6 +333,21 @@ const Ticket = new GQLListSchema('Ticket', {
                 await calculateReopenedCounter(context, existingItem, resolvedData)
             }
 
+            // When creating ticket or updating ticket address,
+            // if client is not passed in resolvedData,
+            // we find a registered user with a phone number that matches the contact's phone number
+            // and an address that matches the ticket address.
+            if (userType === STAFF) {
```

Comment:
It's better not to use the `type` here. You can use `isNull(client)`. Don't forget! We also have a `service` user.type. This logic should work for User with type = `service`

Reply:
checked if user type is not resident and client is null